protocols/imap/src/main/java/org/apache/james/imap/processor/ListProcessor.java

```diff
@@ -148,15 +148,22 @@ protected final void doProcess(String referenceName, String mailboxName, ImapSes
                     basePath = PathConverter.forSession(session).buildFullPath(finalReferencename);
                 }
 
-                results = getMailboxManager().search(
+                if (mailboxName.equals("*")) {
+                    results = getMailboxManager().search(
+                        MailboxQuery.builder()
+                            .matchesAllMailboxNames()
+                            .build(),
+                        mailboxSession);
+                } else {
+                    results = getMailboxManager().search(
```

Comment:
is the test written in the previous commit working without this code?

Reply:
No, it does not work without this changed code

Reply:
so this commit should appear before

Reply:
Or you introduce the test with a @ignore and remove it once you fix it